### PR TITLE
fix kfctl

### PR DESCRIPTION
**Pull request by kunmingg**
kfctl was broken: https://github.com/kubeflow/kfctl/issues/167

replace `gogetter` with http get

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfctl/169)

### CODE REVIEW

**Comment by ryandawsonuk** [comment]
I hit this error so tried to install from this fork. It does resolve the EOF error and I get much further.

Eventually though I hit this:
```
failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: error evaluating kustomization manifest for istio Error absolute path error in 'kustomize/istio' : evalsymlink failure on 'kustomize/istio'
```
I am using https://github.com/ryandawsonuk/seldon-gitops/blob/master/kfctl_istio_dex_noseldon.yaml instead of https://github.com/kubeflow/manifests/blob/master/kfdef/kfctl_istio_dex.yaml but I've compared the yaml on them and the differences are minimal.

**Comment by ryandawsonuk** [comment]
Ok that one must be my problem as I don't get that if I do `kfctl apply -f https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_istio_dex.yaml -V`

I do get `"metadata-ui" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"metadata-ui", "app.kubernetes.io/component":"metadata", "app.kubernetes.io/instance":"metadata-v0.7.0", "app.kubernetes.io/managed-by":"kfctl", "app.kubernetes.io/name":"metadata", "app.kubernetes.io/part-of":"kubeflow", "app.kubernetes.io/version":"v0.7.0", "kustomize.component":"metadata"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable` if I run on an existing cluster. But I guess that's a different issue.

File: pkg/kfconfig/types.go [hunk 1]
```diff
@@ -530,6 +551,54 @@ func (c *KfConfig) SyncCache() error {
 	return nil
 }
 
+func untar(body []byte, cacheDir string) error {
```
Comment (abhi-g) [inline pkg/kfconfig/types.go]: golang has an untar library. Can this be used instead or reimplementing the function..
https://godoc.org/golang.org/x/build/internal/untar

**Pull request merged**